includes/Classifai/Features/AudioTranscriptsGeneration.php

```diff
@@ -348,6 +348,189 @@ public function add_transcription( string $text = '', int $attachment_id = 0 ) {
 		}
 	}
 
+	/**
+	 * Downloads a remote audio file and saves it to a temporary directory within the uploads folder.
+	 *
+	 * This function performs the following:
+	 * 1. Creates a temporary directory under wp-content/uploads/classifai-temp/.
+	 * 2. Downloads the remote file using wp_safe_remote_get().
+	 * 3. Writes the file to the temporary directory.
+	 * 4. Validates that the file is a supported audio type by checking extension and MIME type.
+	 * 5. Uses `finfo` (if available) for an additional MIME type validation.
+	 *
+	 * @param string $url Remote URL to an audio file.
+	 *
+	 * @return string|WP_Error The path to the saved file on success, or WP_Error on failure.
+	 */
+	public static function remote_url_to_path( string $url ) {
+		$upload_dir = wp_upload_dir();
+		$temp_dir   = trailingslashit( $upload_dir['basedir'] ) . 'classifai-temp/';
+
+		if ( ! file_exists( $temp_dir ) ) {
+			wp_mkdir_p( $temp_dir );
+		}
+
+		$response = wp_safe_remote_get(
+			$url,
+			[
+				'timeout' => 10, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout
+			]
+		);
+
+		if ( is_wp_error( $response ) ) {
+			return new WP_Error( 'download_failed', __( 'Failed to download remote file: ', 'classifai' ) . $response->get_error_message() );
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+
+		if ( empty( $body ) ) {
+			return new WP_Error( 'empty_download', __( 'Downloaded file is empty.', 'classifai' ) );
+		}
+
+		$filename = wp_basename( wp_parse_url( $url, PHP_URL_PATH ) );
+
+		if ( empty( $filename ) ) {
+			$filename = 'audio_' . time() . '.tmp';
+		}
+
+		$temp_file_path = $temp_dir . $filename;
+
+		global $wp_filesystem;
+
+		// Initialize the WordPress filesystem.
+		if ( ! $wp_filesystem ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+			WP_Filesystem();
+		}
+
+		$bytes_written = $wp_filesystem->put_contents(
+			$temp_file_path,
+			$body
+		);
+
+		if ( false === $bytes_written ) {
+			return new WP_Error( 'file_write_failed', __( 'Failed to write temporary file.', 'classifai' ) );
+		}
+
+		// Step 5: Validate audio file type
```

Comment:
So I didn't test things yet but can we test the mime type before we upload the file? If I'm following the logic here correctly, if someone were to call this method with a non-audio file, that file will be uploaded but then this check will fail and return an error but doesn't seem like the file will ever be deleted. I'm worried this could be a security hole, allowing someone to upload arbitrary files